From: Robert Luberda <robert@debian.org>
Date: Mon, 12 Sep 2011 20:25:03 +0200
Subject: 02 Lzma support.

Make upx compilable with the latest version of lzma-dev package.
---
 src/compress_lzma.cpp        | 15 ++++++++-------
 src/stub/src/c/Makevars.lzma |  4 ++--
 2 files changed, 10 insertions(+), 9 deletions(-)

diff --git a/src/compress_lzma.cpp b/src/compress_lzma.cpp
index decca78..9206bb0 100644
--- a/src/compress_lzma.cpp
+++ b/src/compress_lzma.cpp
@@ -94,11 +94,11 @@ int compress_lzma_dummy = 0;
 
 
 #if (WITH_LZMA >= 0x461)
-#include "C/7zVersion.h"
+#include "7zVersion.h"
 #if (WITH_LZMA != (0x100 * MY_VER_MAJOR) + (0x10 * (MY_VER_MINOR / 10)) + (MY_VER_MINOR % 10))
 #  error "WITH_LZMA version mismatch"
 #endif
-#include "C/Types.h"
+#include "Types.h"
 static void *cb_alloc(void *, size_t size) {
     return malloc(size);
 }
@@ -232,9 +232,10 @@ error:
 #if (WITH_LZMA >= 0x461)
 #define _7ZIP_ST 1
 #define kLiteralNextStates kLiteralNextStates_enc
-#include "C/LzmaEnc.h"
-#include "C/LzmaEnc.c"
-#include "C/LzFind.c"
+#define _7ZIP_ST 1
+#include "LzmaEnc.h"
+#include "LzmaEnc.c"
+#include "LzFind.c"
 #undef kLiteralNextStates
 #undef kNumFullDistances
 
@@ -566,8 +567,8 @@ error:
 #if (WITH_LZMA >= 0x461)
 
 #undef _LZMA_PROB32
-#include "C/LzmaDec.h"
-#include "C/LzmaDec.c"
+#include "LzmaDec.h"
+#include "LzmaDec.c"
 
 
 int upx_lzma_decompress    ( const upx_bytep src, unsigned  src_len,
diff --git a/src/stub/src/c/Makevars.lzma b/src/stub/src/c/Makevars.lzma
index e55b783..c0cc231 100644
--- a/src/stub/src/c/Makevars.lzma
+++ b/src/stub/src/c/Makevars.lzma
@@ -11,8 +11,8 @@ endif
 endif
 endif
 
-ifneq ($(wildcard $(UPX_LZMADIR)/C/7zVersion.h),)
-UPX_LZMA_VERSION := $(shell perl -ne 'print "0x$$1$$2" if /^\s*\#\s*define\s+MY_VERSION\s+"(\d)\.(\d\d)[\s\w]*"/;' $(UPX_LZMADIR)/C/7zVersion.h)
+ifneq ($(wildcard $(UPX_LZMADIR)/7zVersion.h),)
+UPX_LZMA_VERSION := $(shell perl -ne 'print "0x$$1$$2" if /^\s*\#\s*define\s+MY_VERSION\s+"(\d)\.(\d\d)[\s\w]*"/;' $(UPX_LZMADIR)/7zVersion.h)
 else ifneq ($(wildcard $(UPX_LZMADIR)/CPP/Windows/Time.cpp),)
 UPX_LZMA_VERSION = 0x459
 else ifneq ($(wildcard $(UPX_LZMADIR)/C/7zBuf.h),)
